pageserver/src/tenant.rs

```diff
@@ -109,7 +112,41 @@ pub use crate::tenant::timeline::WalReceiverInfo;
 /// Parts of the `.neon/tenants/<tenant_id>/timelines/<timeline_id>` directory prefix.
 pub const TIMELINES_SEGMENT_NAME: &str = "timelines";
 
-pub const TENANT_ATTACHING_MARKER_FILENAME: &str = "attaching";
+pub const TENANT_ATTACHING_LEGACY_MARKER_FILENAME: &str = "attaching";
+pub const TENANT_ATTACHING_MARKER_SUFFIX: &str = "___attaching";
+
+/// The error message does not include the tenant ID.
+/// The caller is responsible for adding it.
+#[derive(Debug, thiserror::Error)]
+pub enum AttachError {
+    #[error("some local filesystem state already exists: {what}: {path:?}")]
+    LocalStateExists { what: &'static str, path: PathBuf },
+    #[error("{op} marker file {path:?}: {error:#}")]
+    CreateMarkerFile {
+        op: &'static str,
+        path: PathBuf,
+        #[source]
+        error: std::io::Error,
+    },
+    #[error("marker file is missing: {path:?}")]
+    MarkerFileMissing { path: PathBuf },
+}
+
+struct AttachDataForLoad {
+    remote_startup_data: RemoteStartupData,
+    remote_client: RemoteTimelineClient,
+}
+
+enum TenantLoadReason {
+    Regular,
```

Comment:
The `/load` endpoint also uses it. Have a third enum variant for it, just because we can?

Reply:
Pushed commit `differentiated TenantLoadReason::NotAttaching, including differentia...` to address this. Not exactly pretty, but explicit and properly differentiated, so that future readers of the code understand immediately what code paths share the tenant-load code.

Reply:
Aah I see `/load` is a counterpart of `/ignore`? I only remember discussion about adding an `/reload` but I guess this is it.

[07357ef](https://github.com/neondatabase/neon/pull/3466/commits/07357ef0b0e10a52fdd95e66e932151711cc86a2) is the commit.

NotAttach vs. Attach makes me think if an Option would be applicable but perhaps this at least makes one think what other kinds of calls there might be than attach.